src/Module/BaseModeration.php

```diff
@@ -0,0 +1,110 @@
+ * called in the inheriting module.
+ *
+ * Additionally, it puts together the administration page aside with all the administration links.
+ *
+ * @package Friendica\Module
+ */
+abstract class BaseModeration extends BaseModule
+{
+	/**
+	 * Checks moderator access and throws exceptions if not logged-in moderator
+	 *
+	 * @param bool $interactive
+	 * @return void
+	 * @throws HTTPException\ForbiddenException
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	public static function checkModerationAccess(bool $interactive = false)
+	{
+		if (!DI::userSession()->getLocalUserId()) {
+			if ($interactive) {
+				DI::sysmsg()->addNotice(DI::l10n()->t('Please login to continue.'));
+				DI::session()->set('return_path', DI::args()->getQueryString());
+				DI::baseUrl()->redirect('login');
+			} else {
+				throw new HTTPException\UnauthorizedException(DI::l10n()->t('Please login to continue.'));
+			}
+		}
+
+		if (!DI::app()->isSiteAdmin()) {
+			throw new HTTPException\ForbiddenException(DI::l10n()->t('You don\'t have access to administration pages.'));
+		}
+
+		if (DI::userSession()->getSubManagedUserId()) {
+			throw new HTTPException\ForbiddenException(DI::l10n()->t('Submanaged account can\'t access the administration pages. Please log back in as the main account.'));
+		}
+	}
+
+	protected function content(array $request = []): string
+	{
+		self::checkModerationAccess(true);
+
+		// Header stuff
+		DI::page()['htmlhead'] .= Renderer::replaceMacros(Renderer::getMarkupTemplate('moderation/settings_head.tpl'), []);
+
+		/*
+		 * Side bar links
+		 */
+
+		// array(url, name, extra css classes)
+		// not part of $aside to make the template more adjustable
+		$aside_sub = [
+			'information' => [DI::l10n()->t('Information'), [
+				'overview' => ['moderation'             , DI::l10n()->t('Overview')                , 'overview'],
+			]],
+			'configuration' => [DI::l10n()->t('Configuration'), [
+				'users' => ['moderation/users'       , DI::l10n()->t('Users')                   , 'users'],
+			]],
+			'tools' => [DI::l10n()->t('Tools'), [
+				'contactblock' => ['moderation/blocklist/contact', DI::l10n()->t('Contact Blocklist')  , 'contactblock'],
+				'blocklist'    => ['moderation/blocklist/server' , DI::l10n()->t('Server Blocklist')   , 'blocklist'],
+				'deleteitem'   => ['moderation/item/delete' , DI::l10n()->t('Delete Item')             , 'deleteitem'],
+			]],
+			'diagnostics' => [DI::l10n()->t('Diagnostics'), [
+				'itemsource' => ['moderation/item/source' , DI::l10n()->t('Item Source')             , 'itemsource'],
+			]],
+		];
+
+		$t = Renderer::getMarkupTemplate('moderation/aside.tpl');
+		DI::page()['aside'] .= Renderer::replaceMacros($t, [
+			'$subpages'  => $aside_sub,
+			'$admtxt'    => DI::l10n()->t('Moderation'),
+			'$h_pending' => DI::l10n()->t('User registrations waiting for confirmation'),
+			'$modurl'    => 'moderation/'
+		]);
+
+		return '';
```

Comment:
If it doesn't return anything, shouldn't this method be a `rawContent()`?

What do you think @MrPetovan ?

Reply:
No, this is meant as a common method for all the child class `content()` methods. It's a little clunky because we don't use the return value but that's how we've done it for other pages with a similar common display (like the admin and settings modules)